op-defender/psp_executor/defender.go

```diff
@@ -0,0 +1,290 @@
+package psp_executor
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"math/big"
+	"net/http"
+)
+
+// **********************************************************************
+// *                        Informations:                               *
+// **********************************************************************
+// * Sepolia:                                                           *
+// * deputyGuardianSepolia: 0x4220C5deD9dC2C8a8366e684B098094790C72d3c *
+// * SuperChainConfigSepolia: 0xC2Be75506d5724086DEB7245bd260Cc9753911Be *
+// * FoS on Sepolia: 0x837DE453AD5F21E89771e3c06239d8236c0EFd5E        *
+// **********************************************************************
+// * Mainnet:                                                           *
+// * deputyGuardianMainnet: 0x5dC91D01290af474CE21DE14c17335a6dEe4d2a8  *
+// **********************************************************************
+
+const (
+	MetricsNamespace = "psp_executor"
+	SepoliaRPC       = "https://proxyd-l1-consensus.primary.sepolia.prod.oplabs.cloud"
+	MainnetRPC       = "https://proxyd-l1-consensus.primary.mainnet.prod.oplabs.cloud"
+	LocalhostRPC     = "http://localhost:8545"
+)
+
+type Account struct {
+	Address  common.Address
+	Nickname string
+}
+
+type Defender struct {
+	log                     log.Logger
+	port                    string
+	SuperChainConfigAddress string
+	l1Client                *ethclient.Client
+	router                  *mux.Router
+	// metrics
+	latestPspNonce      *prometheus.GaugeVec
+	unexpectedRpcErrors *prometheus.CounterVec
+}
+
+// Define a struct that represents the data structure of your response
+type Response struct {
+	Message string `json:"message"`
+	Status  int    `json:"status"`
+}
+
+type RequestData struct {
+	Pause     bool   `json:"pause"`
+	Timestamp int64  `json:"timestamp"`
+	Operator  string `json:"operator"`
+	Calldata  string `json:"calldata"` //temporary field as the calldata will be fetched from the GCP in the future (so will be removed in the future PR).
+}
+
+// handlePost handles POST requests and processes the JSON body
+func handlePost(w http.ResponseWriter, r *http.Request) {
+	var data RequestData
```

Comment:
If present in the header X-Request-ID or similar, we should get it and use for the logging.
This way if authentication is in place we can track request and link back to the authenticated user that performed the action.

Reply:
Yes, I will add this into the todo to keep tracking of the header thanks!